Mathlib/CategoryTheory/Functor/Derived/LeftDerived.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+import Mathlib.CategoryTheory.Localization.Predicate
+
+/-!
+# Left derived functors
+
+In this file, given a functor `F : C ⥤ H`, and `L : C ⥤ D` that is a
+localization functor for `W : MorphismProperty C`, we define
+`F.totalLeftDerived L W : D ⥤ H` as the right Kan extension of `F`
+along `L`: it is defined if the type class `F.HasLeftDerivedFunctor W`
+asserting the existence of a right Kan extension is satisfied.
+(The name `totalLeftDerived` is to avoid name-collision with
+`Functor.leftDerived` which are the left derived functors in
+the context of abelian categories.)
+
+Given `LF : D ⥤ H` and `α : L ⋙ RF ⟶ F`, we also introduce a type class
+`F.IsLeftDerivedFunctor α W` saying that `α` is a right Kan extension of `F`
+along the localization functor `L`.
+
+(This file was obtained by dualizing the results in the file
+`CategoryTheory.Functor.Derived.RightDerived`.)
+
+## References
+
+* https://ncatlab.org/nlab/show/derived+functor
+
+-/
+
+namespace CategoryTheory
+
+namespace Functor
+
+variable {C C' D D' H H' : Type _} [Category C] [Category C']
+  [Category D] [Category D'] [Category H] [Category H']
+  (LF'' LF' LF : D ⥤ H) {F F' F'' : C ⥤ H} (e : F ≅ F') {L : C ⥤ D}
+  (α'' : L ⋙ LF'' ⟶ F'') (α' : L ⋙ LF' ⟶ F') (α : L ⋙ LF ⟶ F) (α'₂ : L ⋙ LF' ⟶ F)
+  (W : MorphismProperty C)
+
+/-- A functor `LF : D ⥤ H` is a left derived functor of `F : C ⥤ H`
+if it is equipped with a natural transformation `α : L ⋙ LF ⟶ F`
+which makes it a right Kan extension of `F` along `L`,
+where `L : C ⥤ D` is a localization functor for `W : MorphismProperty C`. -/
+class IsLeftDerivedFunctor [L.IsLocalization W] : Prop where
+  isRightKanExtension' : LF.IsRightKanExtension α
+
+lemma IsLeftDerivedFunctor.isRightKanExtension
+    [L.IsLocalization W] [LF.IsLeftDerivedFunctor α W] :
+    LF.IsRightKanExtension α :=
+  IsLeftDerivedFunctor.isRightKanExtension' W
```

Comment:
So I think what Andrew is referring to is the following (in case you didn't already know!):
```suggestion
class IsLeftDerivedFunctor (LF : D ⥤ H) {F : C ⥤ H} {L : C ⥤ D} (α : L ⋙ LF ⟶ F)
    (W : MorphismProperty C) [L.IsLocalization W] : Prop where
  isRightKanExtension (LF α) : LF.IsRightKanExtension α
```
So for this to work you need the explicit arguments `LF` and `α` to appear in the statement (and not as variables). I added other implicit arguments here as well, as otherwise they would appear before `LF` and hence the dot notation would break. I think you should do this in the `RightDerived` file as well

Reply:
Thanks to both of you, I will make the changes soon!